### ISSUE TITLE
Uncaught TypeError: Cannot read property 'current' of undefined

### ISSUE THREAD

**Issue by aminul-technext** [jodit/jodit-react#4]
```
Uncaught TypeError: Cannot read property 'current' of undefined
    at t.calcPathImd (jodit.min.js:10)
    at jodit.min.js:10
```
I have followed this tutorial.
https://xdsoft.net/jodit/examples/intergration/react-jodit.html
and got this error in browser console.

If I block the setState then there is no errors.

**Comment by kernel72** [comment]
I have the same issue. Has anyone found solution/workaround, please?

**Comment by vrielsa** [comment]
I'm not confident if this will help you or not. But the way it was working for me was if I set the configuration as a property of my custom component. By passing the property to JodiEditor the error went away.

`import React, {Component} from 'react';
import {connect} from 'react-redux';
import 'jodit';
import JoditEditor from 'jodit-react';
import { contentChange } from '../Actions/Editor/EditorActions';

@connect((store) => {
    return {
        content: store.editor.content,
    }
})
class Editor extends Component {
    onUpdateContent = (value) => {
        this.props.dispatch(contentChange(value));
    };

    config = {
        readonly: false,
        minHeight: 500
    };

    render() {
        return (
            <JoditEditor
                value={this.props.content}
                config={this.config}
                onChange={this.onUpdateContent}
            />
        )
    }
}

export const EditorContainer = connect((store) => {
    return store;
})(Editor);`

Happy hacking :-)

**Comment by kernel72** [comment]
Well, As w/a I have found solution - not to use this component at all   😃

```javascript

import Jodit from 'jodit'
import 'jodit/build/jodit.min.css'

class JoditWrapper extends Component {

    componentDidMount(){
        const {onContentUpdate, initValue} = this.props;
        this.jodit = new Jodit(this.editorRef);
        this.jodit.value = initValue;
        this.jodit.events.on('change', content => onContentUpdate && onContentUpdate(content))
    }

    render (){
        return (
            <textarea ref={ref => this.editorRef = ref}/>
        )
    }
}

```

**Comment by marchaos** [comment]
Yep - saw this as well and moved my config outside of the component otherwise the Jodit component will render every time that you parent component's render method is called.

**Comment by shivam4786** [comment]
> I'm not confident if this will help you or not. But the way it was working for me was if I set the configuration as a property of my custom component. By passing the property to JodiEditor the error went away.
> 
> ```
> class Editor extends Component {
> 
>     onUpdateContent = (value) => {
>          this.props.dispatch(contentChange(value));
>      };
> 
>     config = {
>         readonly: false,
>         minHeight: 500
>     };
> 
>     render() {
>         return (
>             <JoditEditor
>                  value={this.props.content}
>                  config={this.config}
>                  onChange={this.onUpdateContent}
>             />
>         )
>     }
> }
> ```



> I'm not confident if this will help you or not. But the way it was working for me was if I set the configuration as a property of my custom component. By passing the property to JodiEditor the error went away.
> 
> ```
> class Editor extends Component {
> 
>     onUpdateContent = (value) => {
>          this.props.dispatch(contentChange(value));
>      };
> 
>     config = {
>         readonly: false,
>         minHeight: 500
>     };
> 
>     render() {
>         return (
>             <JoditEditor
>                  value={this.props.content}
>                  config={this.config}
>                  onChange={this.onUpdateContent}
>             />
>         )
>     }
> }
> ```

**Comment by shivam4786** [comment]
> I'm not confident if this will help you or not. But the way it was working for me was if I set the configuration as a property of my custom component. By passing the property to JodiEditor the error went away.
> 
> ```
> class Editor extends Component {
> 
>     onUpdateContent = (value) => {
>          this.props.dispatch(contentChange(value));
>      };
> 
>     config = {
>         readonly: false,
>         minHeight: 500
>     };
> 
>     render() {
>         return (
>             <JoditEditor
>                  value={this.props.content}
>                  config={this.config}
>                  onChange={this.onUpdateContent}
>             />
>         )
>     }
> }
> ```

This Worked like Charm, Thanks Man

**Comment by cunamy** [comment]
In JoditEditor.js has error, need change in useEffect
`useEffect(() => {
		const blurHandler = value => {
			onBlur && onBlur(value)
		};

		const changeHandler = value => {
			onChange && onChange(value)
		};

		textArea.current = new Jodit(textArea.current, config);

		textArea.current.value = value;
		textArea.current.events.on('blur', () => blurHandler(textArea.current.value));
		textArea.current.events.on('change', () => changeHandler(textArea.current.value));
		textArea.current.workplace.tabIndex = tabIndex || -1;
		**const elem = textArea.current.element;
		return () => {
			textArea.current.destruct();
      		textArea.current = elem;**
		}
	}, [config]);`

**Comment by NicolasBrinkhus** [comment]
I had the same issue, follow what worked for me. 

`import React, {useState, useMemo} from 'react';
import JoditEditor from "jodit-react";

function getConfig(props) {
config = {
  readonly: false
}
return config;
}

const Example = (props) => {
const [content, setContent] = useState('')
const newConfig = useMemo(() => getConfig(props), [props]); 

return (
  <JoditEditor
    value={content}
    config={newConfig}
    onBlur={newContent => setContent(newContent)}
    onChange={newContent => {}}
    />
  );
}`

**Comment by HarishSha** [comment]
#### I ended up writing wrapper of `jodit` to fix issues. 

```js
import React, { useEffect, useRef, forwardRef, useLayoutEffect } from "react";
import PropTypes from "prop-types";
import Jodit from "jodit";
import "jodit/build/jodit.min.css";

const JoditEditor = forwardRef(
  ({ value, config, onChange, onBlur, tabIndex, name }, ref) => {
    const textArea = useRef(null);

    useLayoutEffect(() => {
      if (ref) {
        if (typeof ref === "function") {
          ref(textArea.current);
        } else {
          ref.current = textArea.current;
        }
      }
    }, [textArea]);

    useEffect(() => {
      const blurHandler = value => {
        onBlur && onBlur(value);
      };

      const changeHandler = value => {
        onChange && onChange(value);
      };

      const element = textArea.current;
      textArea.current = Jodit.make(element, config);
      textArea.current.value = value;

      textArea.current.events.on("blur", () =>
        blurHandler(textArea.current.value)
      );
      textArea.current.events.on("change", () =>
        changeHandler(textArea.current.value)
      );

      textArea.current.workplace.tabIndex = tabIndex || -1;
    }, []);

    return <textarea ref={textArea} name={name}></textarea>;
  }
);

JoditEditor.propTypes = {
  value: PropTypes.string,
  tabIndex: PropTypes.number,
  config: PropTypes.object,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
};

export default JoditEditor;
